Replicator/Replicator.cc

```diff
@@ -777,21 +744,27 @@ namespace litecore { namespace repl {
                 bool valid = sub.checkpointer->validateWith(remoteCheckpoint);
                 if (!valid && sub.pusher)
                     sub.pusher->checkpointIsInvalid();
-
+#ifdef TEMP_WORK_AROUND
+            }
+#endif
                 // Now we have the checkpoints! Time to start replicating:
                 startReplicating(coll);
-            }
+#ifndef TEMP_WORK_AROUND
+        }
+#endif
```

Comment:
Probably I got confused with the ifdef stuff. But the `startReplicating(coll)` should be done only when `refresh` is `false`.

Reply:
It is to prevent the active replicator to send message before it receiving the response of getCheckpoint. Currently, if the local checkpoint is available, we startReplicating() without waiting for the response of getCheckpoint. I gather that it is an optimization and 3.0 protocol does not provide that the first message being a handshake, getCollections. Without the handshake, the server won't be able to interpret the collection index. I temporarily disable this optimization, that is, the active replicator start replicating after the first message is received. Of course, I mean to remove it later to be a "genuine" 3.0 client.